addon/ng2/tasks/install.js

```diff
@@ -0,0 +1,178 @@
+/* jshint node: true, esnext: true */
+'use strict';
+
+const Promise     = require('ember-cli/lib/ext/promise');
+const Task        = require('ember-cli/lib/models/task');
+const chalk       = require('chalk');
+const path        = require('path');
+const fs          = require('fs');
+const fse         = require('fs-extra');
+const exec        = Promise.denodeify(require('child_process').exec);
+const packageJSON = path.resolve(process.cwd(), 'package.json');
+const nodeModules = path.resolve(process.cwd(), 'node_modules');
+
+module.exports = Task.extend({
+  command: '',
+  completionOKMessage: 'Successfully installed.',
+  completionErrorMessage: 'Error installing package.',
+
+  run: function(options) {
+    this.package = options.package;
+    this.typings = options.typings;
+
+    this.disableLogger();
+
+    this.ui.startProgress(chalk.green('Installing package:', chalk.yellow(this.package)), chalk.green('.'));
+
+    return this.installProcedure();
+  },
+
+  installProcedure: function() {
+    const that = this;
+
+    return that.npmInstall(that.package)
+    .then(() => {
+      return that.storeInPackageJSON(that.package)
+      .then(() => {
+        return that.copyPackages()
+        .then(() => {
+          that.ui.stopProgress();
+          if (that.typings) {
+            that.ui.startProgress(chalk.green('Installing typings:', chalk.yellow(that.typings)), chalk.green('.'));
+            return that.typingsInstall(that.typings)
+            .then(() => {
+              that.ui.stopProgress();
+              that.announceOKCompletion();
+            });
+          }
+          else {
+            that.announceOKCompletion();
+          }
+        });
+      });
+    });
+  },
+
+  npmInstall: function(pkg) {
+    let cmd = ['npm install'];
+    const npmOptions = [
+      '--loglevel error',
+      '--color always',
+      '--save-dev'
+    ];
+    cmd.push(pkg);
+
+    return exec(cmd.concat(npmOptions).join(' '));
+  },
+
+  typingsInstall: function(pkg) {
+    let cmd = ['typings install'];
+    let options = [
+      '--ambient',
+      '--save'
+    ];
+    cmd.push(pkg);
+
+    return exec(cmd.concat(options).join(' '));
+  },
+
+  storeInPackageJSON: function(pkg) {
```

Comment:
I would prefer single config file, but can make a separate one if you think this should be more transparent. Otherwise is not a big deal having a large JSON file the users should not update it directly anyway.